### PR TITLE
feat(FTDAS): Add Delete /todos/completed to delete completed tasks

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
### Description of Change
- Added Delete /todos/completed api at controller
- Added delete functionality at providers and repository
### Associated Ticket Link:
- https://trello.com/c/X4lhWrvc

### CODE REVIEW

**Comment by ragib-hossain-sazim** [comment]
Just noticed one thing you have been using FTDAS inside commit message for all server prs. This should be FTDA, the name of the product. We follow the abbreviation of the product name not the repo name inside commit message. Not a biggy but please keep that in mind for your future commits.

File: src/tasks/tasks.repository.ts [hunk 1]
```diff
@@ -17,4 +17,9 @@ export class TaskRepository extends EntityRepository<Task> {
     }
     return task;
   }
+
+  async deleteCompletedTasks(): Promise<void> {
+    const completedTasks = await this.find({ status: 'completed' });
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.repository.ts:22]: move `find` inside `removeCompletedTasks` of `service` file.

File: src/tasks/tasks.controller.ts [hunk 2]
```diff
@@ -38,4 +38,14 @@ export class TasksController {
...
   }
+
+  @Get(':id')
+  async findOneTask(@Param('id') id: string): Promise<Task> {
+    return this.tasksService.findOne(+id);
+  }
+
+  @Delete('completed')
+  async removeCompletedTasks(): Promise<void> {
+    await this.tasksService.removeCompletedTasks();
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.controller.ts:49]:
```suggestion
    await this.tasksService.removeCompleted();
```

File: src/tasks/tasks.service.ts [hunk 3]
```diff
@@ -1,41 +1,39 @@
...
-    }
+  async findOne(id: number): Promise<Task> {
+    return await this.taskRepository.findOneOrFail(id);
+  }
+
+  async removeCompletedTasks(): Promise<void> {
+    const completedTasks = await this.taskRepository.find({
+      status: 'completed',
+    });
+    await this.taskRepository.removeCompletedTasks(completedTasks);
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.service.ts:37]:
```suggestion
    await this.tasksRepository.removeCompleted(completedTasks);
```

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments.

**Review by ragib-hossain-sazim — APPROVED** [review]

**Comment by ragib-hossain-sazim** [comment]
Please resolve merge conflicts before merging

**Pull request merged**